src/main/java/org/opentripplanner/graph_builder/module/StreetLinkerModule.java

```diff
@@ -257,6 +260,30 @@ private void linkTransitEntrances(Graph graph) {
     }
   }
 
+  private void linkStationCentroids(Graph graph) {
+    LOG.info("Linking instances of {} to graph...", StationCentroidVertex.class.getSimpleName());
+    for (StationCentroidVertex tVertex : graph.getVerticesOfType(StationCentroidVertex.class)) {
+      graph
+        .getLinker()
+        .linkVertexPermanently(
+          tVertex,
+          new TraverseModeSet(TraverseMode.WALK),
+          LinkingDirection.BOTH_WAYS,
+          (vertex, streetVertex) ->
+            List.of(
+              StreetStationCentroidLink.createStreetStationLink(
+                (StationCentroidVertex) vertex,
+                streetVertex
+              ),
+              StreetStationCentroidLink.createStreetStationLink(
+                streetVertex,
+                (StationCentroidVertex) vertex
+              )
+            )
+        );
+    }
```

Comment:
I agree about the naming. I'll fix that.

The second point is harder to find a perfect solution to IMO. I agree that complicated inline functions makes it hard to follow.

But I'm also a bit reluctant about moving logic to very shallow and specific private methods that are only used in one place. Because it clutters the internal structure of the class and makes it more difficult to get an overview of it. And also that I probably have to go look at the method anyways in order to understand what is happening, and then it's nicer to have Locality of Behaviour.

What about having the lambda as a locally named variable. Do you think that is as bad?

Reply:
I have not hear about "Locality of Behaviour" before, but it make sense - especially in dynamic languages. A lambda variable is ok with me - they are not great but ok. There is probably something we are missing here - I think you also agree that this is not easy to read. Personally I am not afraid to make private methods that are used once - they can be used to reveal intention. You could also put the entire
```
   graph
        .getLinker()
        .linkVertexPermanently(
          tVertex,
          new TraverseModeSet(TraverseMode.WALK),
          LinkingDirection.BOTH_WAYS,
          (vertex, streetVertex) ->...
```
into a method. Sorry, I think " also that I probably have to go look at the method anyway" is not true. Most developer spend a great deal of time reading code - often skipping details to find the right place. Being able to get an overview of what is going on is more important than having all detail in one place.

To me there are to many nested levels in the method, if you find a way to break it up - then I am happy.